File: python/src/ttio/value_range.py
"""``ValueRange`` — inclusive numeric interval used for axis bounds."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True, slots=True)
class ValueRange:
    """An inclusive numeric interval ``[minimum, maximum]``.

    Immutable value class. Empty or inverted ranges are permitted at
    construction time; consumers decide whether ``minimum > maximum``
    is meaningful in their context.

    Parameters
    ----------
    minimum : float
        Lower bound (inclusive).
    maximum : float
        Upper bound (inclusive).

    Attributes
    ----------
    span : float
        Difference ``maximum - minimum``.

    Notes
    -----
    API status: Stable.

    Cross-language equivalents
    --------------------------
    Objective-C: ``TTIOValueRange`` · Java:
    ``global.thalion.ttio.ValueRange``.
    """

    minimum: float
    maximum: float

    def contains(self, value: float) -> bool:
        """Return ``True`` if ``value`` lies within the closed range.

        Parameters
        ----------
        value : float

        Returns
        -------
        bool
        """
        return self.minimum <= value <= self.maximum

    @property
    def span(self) -> float:
        """Difference between ``maximum`` and ``minimum``."""
        return self.maximum - self.minimum

File: java/src/main/java/com/dtwthalion/ttio/ValueRange.java
/*
 * TTI-O Java Implementation
 * Copyright (C) 2026 DTW-Thalion
 * SPDX-License-Identifier: LGPL-3.0-or-later
 */
package com.dtwthalion.ttio;

/**
 * Closed numeric range {@code [minimum, maximum]}. Immutable value
 * class.
 *
 * <p>Used by {@link AxisDescriptor} to describe the bounds of a
 * signal axis.</p>
 *
 * <p><b>API status:</b> Stable.</p>
 *
 * <p><b>Cross-language equivalents:</b> Objective-C
 * {@code TTIOValueRange}, Python
 * {@code ttio.value_range.ValueRange}.</p>
 *
 * @since 0.6
 */
public record ValueRange(double minimum, double maximum) {

    /** @return difference between {@code maximum} and {@code minimum}. */
    public double span() {
        return maximum - minimum;
    }

    /** @return {@code true} if {@code value} lies within the closed range. */
    public boolean contains(double value) {
        return minimum <= value && value <= maximum;
    }
}
